.github/workflows/pull-request-checks.yml

```diff
@@ -0,0 +1,323 @@
+        run: chmod +x gradlew
+
+      - uses: actions/cache@v4
+        with:
+          path: |
+            ~/.gradle/caches
+          key: ${{ runner.os }}-gradle-solrj-${{ hashFiles('versions.lock') }}
+          restore-keys: |
+            ${{ runner.os }}-gradle-solrj-
+            ${{ runner.os }}-gradle-
+
+      - name: Test the SolrJ Package
+        run: ./gradlew solr:solrj:test
+
+  # Script tests
+  script-test:
+    name: Run Solr Script Tests
+
+    runs-on: ubuntu-latest
+    # Run this job after pre-commit-checks were successful.
+    needs: pre-commit-checks
+
+    # Run this job always, even if any of the previous jobs were skipped.
+    # This job is executed only if the pre-commit-checks job is successful.
+    # Run this job only if script-related files were modified, that is, when label present.
+    if: always() &&
+      needs.pre-commit-checks.result == 'success' &&
+      contains(github.event.pull_request.labels.*.name, 'start-scripts')
+
+    env:
+      DEVELOCITY_ACCESS_KEY: ${{ secrets.GE_ACCESS_TOKEN }}
+
+    steps:
+      # Setup
+      - uses: actions/checkout@v4
+        with:
+          ref: ${{ github.head_ref }}
+
+      - name: Set up JDK
+        uses: actions/setup-java@v4
+        with:
+          distribution: 'temurin'
+          java-version: 21
+          java-package: jdk
+
+      - name: Setup Gradle
+        uses: gradle/actions/setup-gradle@v4
+
+      - name: Grant execute permission for gradlew
+        run: chmod +x gradlew
+
+      - uses: actions/cache@v4
+        with:
+          path: |
+            ~/.gradle/caches
+          key: ${{ runner.os }}-gradle-binsolr-${{ hashFiles('versions.lock') }}
+          restore-keys: |
+            ${{ runner.os }}-gradle-binsolr-
+            ${{ runner.os }}-gradle-
+
+      - name: Test the bin/solr script
+        run: ./gradlew integrationTests
+
+      - name: Archive logs
+        if: ${{ failure() }}
+        uses: actions/upload-artifact@v4
+        with:
+          name: logs
+          path: solr/packaging/build/test-output
+
+  crave-test:
+    name: Run Solr Tests using Crave.io resources
+
+    runs-on: self-hosted
+    # Run this job after pre-commit-checks were successful.
+    needs: pre-commit-checks
+
+    # Run this job always, even if any of the previous jobs were skipped.
+    # This job is executed only if the pre-commit-checks job is successful.
+    if: always() && needs.pre-commit-checks.result == 'success'
+
+    steps:
+      - name: Destroy previous clone
+        run: crave clone destroy -y /crave-devspaces/pipeline/runs/${GITHUB_RUN_ID}_${GITHUB_RUN_NUMBER} || echo "Clone did not exist"
+        continue-on-error: true
+      - name: Crave clone sources
+        run: crave clone create --projectID 39 /crave-devspaces/pipeline/runs/${GITHUB_RUN_ID}_${GITHUB_RUN_NUMBER}
+      - name: Checkout the correct branch
+        # TODO See if below run checks out latest commit
```

Comment:
This is a TODO that I have not beeing able to resolve yet due to the lack of information.